web_infrastructure/nginx_status_facts.py

```diff
@@ -0,0 +1,148 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible. If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: nginx_status_facts
+short_description: Retrieve nginx status facts.
+description:
+  - Gathers facts from nginx from an URL having C(stub_status) enabled.
+version_added: "2.3"
+author: "René Moser (@resmo)"
+options:
+  url:
+    description:
+      - URL of the nginx status.
+    required: true
+requirements:
+  - "nginx > 1.3.14"
+
+notes:
+  - See http://nginx.org/en/docs/http/ngx_http_stub_status_module.html for more information.
+'''
+
+EXAMPLES = '''
+# Gather status facts from nginx on localhost
+- name: get current http stats
+  nginx_status_facts: url=http://localhost/nginx_status
+'''
+
+RETURN = '''
+---
+nginx_status_facts.active_connections:
+  description: Active connections.
+  returned: success
+  type: int
+  sample: 2340
+nginx_status_facts.accepts:
+  description: The total number of accepted client connections.
+  returned: success
+  type: int
+  sample: 81769947
+nginx_status_facts.handled:
+  description: The total number of handled connections. Generally, the parameter value is the same as accepts unless some resource limits have been reached.
+  returned: success
+  type: int
+  sample: 81769947
+nginx_status_facts.requests:
+  description: The total number of client requests.
+  returned: success
+  type: int
+  sample: 144332345
+nginx_status_facts.reading:
+  description: The current number of connections where nginx is reading the request header.
+  returned: success
+  type: int
+  sample: 0
+nginx_status_facts.writing:
+  description: The current number of connections where nginx is writing the response back to the client.
+  returned: success
+  type: int
+  sample: 241
+nginx_status_facts.waiting:
+  description: The current number of idle client connections waiting for a request.
+  returned: success
+  type: int
+  sample: 2092
+nginx_status_facts.data:
+  description: HTTP response as is.
+  returned: success
+  type: string
+  sample: "Active connections: 2340 \nserver accepts handled requests\n 81769947 81769947 144332345 \nReading: 0 Writing: 241 Waiting: 2092 \n"
+'''
+
+import re
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.urls import fetch_url
+
+
+class NginxStatusFacts(object):
+
+    def __init__(self):
+        self.url = module.params.get('url')
+
+    def run(self):
+        result = {
+            'nginx_status_facts': {
+                'active_connections': None,
+                'accepts': None,
+                'handled': None,
+                'requests': None,
+                'reading': None,
+                'writing': None,
+                'waiting': None,
+                'data': None,
+            }
+        }
+        (response, info) = fetch_url(module, self.url, force=True)
```

Comment:
fetch_url has default timeout of 10 seconds, which might be too long for cases such as downloading stub status. Have you considered adding configurable timeout to this module?